### DISCUSSION TITLE
Permadeath (or any other respawn mode) should be linked to a save file

### DISCUSSION THREAD

**Discussion by ilmarilahti** [FakeFishGames/Barotrauma#14554] [Ideas and suggestions]
The MP host has to remember to re-choose permadeath mode every time when loading up a save, personally I forget it more often than not. (Especially easy to miss when the respawn settings box is scrolled down, I'm not sure when or why exactly it sometimes is and sometimes isn't.) The issue happens with other respawn modes too, though the wrong choice for a single round doesn't feel as jarring with them.

While I appreciate the possibility to switch the mode per round if there's ever need (= I'm not suggesting to remove that option), it's still most often the case that we refer to the entire campaign a "permadeath game" if permadeath is used at all. Not saving the choice almost feels like a bug, as the other respawn settings do get saved. Or does it just remember the previous ones used?

Alternative implementation suggestions:
- Save the respawn mode so that when a save is loaded, the respawn mode that was saved gets chosen. _However_, currently this has the issue that clicking load instantly starts the loaded game, so you would not know about the saved respawn mode until the game was already starting. Perhaps the saved options (maybe including characters, etc) could be loaded into the lobby screen when clicking a save file, so you could "preview" that save?
- **Simpler alternative**: save the respawn mode with the save file, but instead of automatically choosing it, use it only to notify (eg. a dialog box with "continue" and "cancel") the user if they have clicked Load on a save file while having a different respawn mode selected than what is associated with the save file. This would fix this issue without needing further changes to how the lobby currently works.

**Comment by Ek-Videogames** [comment]
Yeah definitely seems like something that should remember from the save once you load it, 
could have it still be switchable in the campaign lobby before launching the round as an option though.

**Comment by Rokvach** [comment]
Tested this, didn't spot any issues. Coming in the next update. Closing discussion.
